### DISCUSSION TITLE
See only sensors on VP18cCOM with USB to RS485 b converter

### DISCUSSION THREAD

**Discussion by GruenerVeltliner** [veista/nilan#146] [Q&A] [answered]
Hi @veista,

first of all thank you for your effort for this Nilan integration.
I tried it also today (v1.2.16) and get it already work.

Nilan VP18cCOM
Version 1.02
Software 2.11.g
ModBus Vers. 2

HA on Raspberry Pi 4
Core.   2024.9.1
Supervisor. 2024.09.1
Operating System. 13.1
Frontend. 20240906.0

But I haven´t any control functions, only sensors.
<img width="953" alt="Bildschirmfoto 2024-09-16 um 20 16 38" src="https://github.com/user-attachments/assets/48a64ec8-7412-4389-8230-93aad2f88c33">
<img width="928" alt="Bildschirmfoto 2024-09-16 um 20 16 51" src="https://github.com/user-attachments/assets/34f1f3e2-9971-43b0-906d-c085806cb818">

This is the converter which I use:
[https://www.waveshare.com/usb-to-rs485-b.htm]

In the Log I didn´t find any issue regarding to the Nilan integration.

![unnamed](https://github.com/user-attachments/assets/f4dc0bde-3366-438d-b096-9c6b512508df)
![unnamed](https://github.com/user-attachments/assets/7b31eb0c-7e8e-4ee4-8394-8a6c38222160)
![unnamed](https://github.com/user-attachments/assets/d2b3e1f7-169a-4dfa-bbbf-67ae2af02678)
![IMG_4338](https://github.com/user-attachments/assets/fcf6d28a-ddb3-457d-9314-56aff0aab9c3)

Do you have any idea what is missing or problem?

Thank you for your answer and help.

Best regards

**Comment by veista** [comment]
Hi. I'm not anywhere near a computer for two weeks, but I have a lot of good guesses why this happened. Your device's modbus version is small and it is missing some component needed for basic controls. Modbus version 2 has very limited functionality so don't get your hopes up, but I'm fairly sure we can find a workaround. It might be just a typo in the config.

**Reply by GruenerVeltliner** [reply]
Thank you for your fast reply. I´m not in hurry with this old device from 2011 ;-).

Yeah, I have read it in the past discussions, that my version would be very limited in functions.
The only thing I just want to control, is the hot water production from the Nilan device. Because now the compressor is always running when the temperature decrease 3°C. My wish is, that I can avoid this during the night.

Just one more question, because you mention that in the config is something wrong. I didn´t type anything in the config data at the custom component.yaml. May this is the problem, that I forgot something to config?

**Comment by veista** [comment, marked as answer]
I checked. The minimum bus version for controls is 3. You will need to get an firmware update from nilan for your device. Then it this integration should work.

If you are able to use usb, there is an integration where you can emulate button presses. I have disabled that functionality due to unsolvable issues. You can try that one instead.

**Reply by GruenerVeltliner** [reply]
Ok, thank you. Then it is clear for me. 
I´ve asked the Nilan service already for a software update. So just wait what they answer.

Two years ago I ask also for the retrofit of the lan controller, but this wasn´t available for my device. I must change the electronics, said the Nilan service to me.

**Reply by veista** [reply]
No, I'm refering to the device map in this integration.

**Reply by GruenerVeltliner** [reply]
With a fantastic support from a german company, i was able to do the software update on my own.
I get now the 2.38t with Modbus Version 19 and now i can also control my device. I came from v 2.11.g

The luck for me was, that i have one of the newer electronics with a bigger processor which is mandatory for an successful update. Otherwise the electronics can be destroyed.

If someone also need help, i can recommend this company:

E²S Energieeffiziente Systeme
Theißensteiner Str. 2a
96472 Rödental
[www.e2s-shop.de](http://www.e2s-shop.de/)

<img width="1180" height="820" alt="IMG_0170" src="https://github.com/user-attachments/assets/8679932b-518f-48df-a275-67e89fca1af6" />

@veista thank you for your work on this integration and help.
